webapp/src/pages/UserStatsPage/UserStatsPage.tsx

```diff
@@ -1,89 +1,115 @@
-          : game.playerTwoScore,
-    };
-    const other: { username?: string; score: number } = {
-      username:
-        username === game.playerOneUsername
-          ? game.playerTwoUsername
-          : game.playerOneUsername,
-      score:
-        username === game.playerOneUsername
-          ? game.playerTwoScore
-          : game.playerOneScore,
-    };
-    const getStringResult = () => {
-      return self.score < other.score
-        ? 'losing'
-        : self.score === other.score
-        ? 'tying'
-        : 'winning';
-    };
-    return {
-      title:
-        `'${self.username}' scored ${self.score} points VS ` +
-        `'${other.username}' that scored ${
-          other.score
-        } points, ${getStringResult()} the game`,
-      subtitle: `a ${game.gameMode} game ran for ${game.gameDurationInSeconds} seconds`,
-      key: game.id,
-      altText: `Game started on ${game.createdAt}`,
-    };
-  };
-
-  const getUserGames = useCallback(
-    <T extends Query>(requestParameters: T) =>
-      gameApi.gameControllerGetUserGames({
-        ...requestParameters,
-        limit: ENTRIES_LIMIT,
-        userName: username!,
-      } as GameControllerGetUserGamesRequest),
+  const { data, isLoading } = useData<UserLevelWithTimestamp[]>(getData);
+  const getStats = useCallback(
+    () =>
+      gameApi.gameControllerGetUserStats({
+        username: username!,
+        mode: GameControllerGetUserLevelHistoryModeEnum.Classic,
+      }),
     [username],
   );
-
+  const { data: stats, isLoading: areStatsLoading } =
+    useData<GameStats>(getStats);
+  const getUserByUserName = useCallback(
+    () => usersApi.userControllerGetUserByUserName({ userName: username! }),
+    [username],
+  );
+  const { data: user, isLoading: isUserLoading } = useData(getUserByUserName);
+  if (isLoading || isUserLoading || areStatsLoading) return <LoadingPage />;
+  if (!data || !user || !stats) return <NotFoundPage />;
+  data.unshift({
+    username: user.username,
+    timestamp: user.createdAt,
+    level: 1,
+    gameId: '', // maybe we should allow null in gameId here,
+  });
   return (
-    <RowsPageTemplate
-      title="games played"
-      isLoading={isUserLoading}
-      isNotFound={!user}
-      fetchFn={getUserGames}
-      dataMapper={mapGameToRow}
-      avatarProps={{
-        url: user ? `${AVATAR_EP_URL}/${user.avatarId}` : '',
-        XCoordinate: user ? user.avatarX : 0,
-        YCoordinate: user ? user.avatarY : 0,
-      }}
-      avatarLabel={username!}
-      avatarCaption={`level ${user?.level ?? 1}`}
-      avatarLinkTo={`${USER_URL}/${username!}`}
-    />
+    <div className="stats-page">
+      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        {`game stats for '${user.username}'`}
+      </Header>
+      <div className="graphs">
+        <div className="line-graph-wrapper">
+          <Line
+            data={{
+              labels: data.map((item) => item.timestamp),
```

Comment:
we could customize these colors, but tbh, I like the default ones, so I'm not changing them.